File: ProjectSpace/Android/app/src/main/java/dream/security/msgEncryptor/EncryptTool.java
package dream.security.msgEncryptor;

import java.io.UnsupportedEncodingException;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.Arrays;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;
import android.util.Base64;
public class EncryptTool extends Object {

    private static SecretKeySpec secretKey;
    private static byte[] key;
    private static String noErr = "fine";
    private static String decryptedString;
    private static String encryptedString;

    public static void setKey(String myKey) {
        MessageDigest sha = null;
        try {
            key = myKey.getBytes("UTF-8");
            sha = MessageDigest.getInstance("SHA-1");
            key = sha.digest(key);
            key = Arrays.copyOf(key, 16); // use only first 128 bit
            //System.out.println(new String(key,"UTF-8"));
            secretKey = new SecretKeySpec(key, "AES");
        } catch (NoSuchAlgorithmException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        } catch (UnsupportedEncodingException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
    }

    public static String getDecryptedString() {
        return decryptedString;
    }

    public static void setDecryptedString(String decryptedString) {
        EncryptTool.decryptedString = decryptedString;
    }

    public static String getEncryptedString() {
        return encryptedString;
    }

    public static void setEncryptedString(String encryptedString) {
        EncryptTool.encryptedString = encryptedString;
    }

    public static String encrypt(String strToEncrypt) {
        try {
            Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
            cipher.init(Cipher.ENCRYPT_MODE, secretKey);
            byte[] passin = cipher.doFinal(strToEncrypt.getBytes("UTF-8"));
            setEncryptedString(Base64.encodeToString(passin, 0));
        } catch (Exception e) {
            System.out.println("Error while encrypting: " + e.toString());
        }
        return null;
    }

    public static void decrypt(String strToDecrypt) {
        try {
            Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5PADDING");
            cipher.init(Cipher.DECRYPT_MODE, secretKey);
            byte[] passin = strToDecrypt.getBytes();
            EncryptTool.setDecryptedString(new String(cipher.doFinal(Base64.decode(passin, 0))));
        } catch (Exception e) {
            System.out.println("Error while decrypting: " + e.toString());
            noErr = "ERR:WRONG_PW";
        }
    }

    public static String aesEncrypt(String contents, String pw) {
        final String strToEncrypt = contents;
        final String strPssword = pw;
        setKey(strPssword);
        encrypt(strToEncrypt.trim());
        String encrypted = EncryptTool.getEncryptedString();
        return encrypted;
    }

    public static String aesDecrypt(String contents, String pw) {
        noErr = "fine";
        final String strPssword = pw;
        setKey(strPssword);
        final String strToDecrypt = contents;
        decrypt(strToDecrypt.trim());
        String decrypted = EncryptTool.getDecryptedString();
        if (noErr.equals("fine")) {
            return decrypted;
        } else {
            return noErr;
        }
    }
}

File: MobileProjectSpace/Kotlin/app/src/main/java/dream/security/msgEncryptor/EncryptTool.kt
package dream.security.msgEncryptor

import android.os.Build
import android.support.annotation.RequiresApi
import java.io.UnsupportedEncodingException
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException
import java.util.Arrays

import javax.crypto.Cipher
import javax.crypto.spec.SecretKeySpec
import android.util.Base64

class EncryptTool : Any() {
    companion object {

        private var secretKey: SecretKeySpec? = null
        private var key: ByteArray? = null
        private var noErr = "fine"
        var decryptedString: String? = null
        var encryptedString: String? = null

        @RequiresApi(Build.VERSION_CODES.GINGERBREAD)
        fun setKey(myKey: String) {
            var sha: MessageDigest? = null
            try {
                sha = MessageDigest.getInstance("SHA-1")
                key = sha!!.digest(key)
                key = Arrays.copyOf(key!!, 16) // use only first 128 bit
                //System.out.println(new String(key,"UTF-8"));
                secretKey = SecretKeySpec(key, "AES")
            } catch (e: NoSuchAlgorithmException) {
                // TODO Auto-generated catch block
                e.printStackTrace()
            } catch (e: UnsupportedEncodingException) {
                // TODO Auto-generated catch block
                e.printStackTrace()
            }

        }

        fun encrypt(strToEncrypt: String): String? {
            try {
                val cipher = Cipher.getInstance("AES/ECB/PKCS5Padding")
                cipher.init(Cipher.ENCRYPT_MODE, secretKey)
                val passin = cipher.doFinal(strToEncrypt.toByteArray(charset("UTF-8")))
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.FROYO) {
                    encryptedString = Base64.encodeToString(passin, 0)
                }
            } catch (e: Exception) {

            }

            return null
        }

        fun decrypt(strToDecrypt: String) {
            try {
                val cipher = Cipher.getInstance("AES/ECB/PKCS5PADDING")
                cipher.init(Cipher.DECRYPT_MODE, secretKey)
                val passin = strToDecrypt.toByteArray()
                EncryptTool.decryptedString = String(cipher.doFinal(Base64.decode(passin, 0)))
            } catch (e: Exception) {
                noErr = "ERR:WRONG_PW"
            }

        }

        fun aesEncrypt(contents: String, pw: String): String? {
            setKey(pw)
            encrypt(contents.trim { it <= ' ' })
            return encryptedString
        }

        fun aesDecrypt(contents: String, pw: String): String? {
            noErr = "fine"
            setKey(pw)
            decrypt(contents.trim { it <= ' ' })
            val decrypted = EncryptTool.decryptedString
            return if (noErr == "fine") {
                decrypted
            } else {
                noErr
            }
        }
    }
}